redfish-core/lib/log_services.hpp

```diff
@@ -3346,4 +3407,983 @@ inline void requestRoutesPostCodesEntry(App& app)
+                                      std::next(redfishUriIt));
+            }
+            else
+            {
+                std::string redfishUri = std::get<1>(*redfishUriIt) + "/" +
+                                         getIsolatedHwItemId(dbusObjPath);
+                if (redfishUri.find("<str>") == std::string::npos)
+                {
+                    nlohmann::json& entryJson =
+                        (entryJsonIdx > 0
+                             ? asyncResp->res
+                                   .jsonValue["Members"][entryJsonIdx - 1]
+                             : asyncResp->res.jsonValue);
+
+                    entryJson["Links"]["OriginOfCondition"] = {
+                        {"@odata.id", redfishUri}};
+                }
+                else
+                {
+                    /**
+                     * For example,
+                     * "/redfish/v1/Systems/system/Processors/<str>/ \
+                     * SubProcessors/core0"
+                     */
+                    fillRedfishUriWithIds(asyncResp, entryJsonIdx, dbusObjPath,
+                                          redfishUri);
+                }
+            }
+        },
+        "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetObject", dbusObjPath.str,
+        std::get<0>(*redfishUriIt));
+}
+
+/**
+ * @brief API used to get redfish uri of the given dbus object and fill into
+ *        "OriginOfCondition" property of LogEntry schema.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ *
+ * @return The redfish response with "OriginOfCondition" property of
+ *         LogEntry schema if success else nothing in redfish response.
+ */
+inline void getRedfishUriByDbusObjPath(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const sdbusplus::message::object_path& dbusObjPath,
+    const size_t entryJsonIdx)
+{
+    getRedfishUriFromList(asyncResp, entryJsonIdx, dbusObjPath,
+                          redfishUriList.begin());
+}
+
+/**
+ * @brief API used to get "PrettyName" by using the given dbus object path
+ *        and fill into "Message" property of LogEntry schema.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details in the appropriate entry json object.
+ *
+ * @return The redfish response with "Message" property of LogEntry schema
+ *         if success else nothing in redfish response.
+ */
+
+inline void getPrettyNameByDbusObjPath(
```

Comment:
This is the way upstream is headed: https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/43967/18/redfish-core/include/utils/name_utils.hpp 
Anyway we can pull that in and use that?

Reply:
I will update after tested the pointed code - Here, I am playing with a json object.